diff -ruN cppzmq-4.7.1/zmq_addon.hpp cppzmq/zmq_addon.hpp
--- cppzmq-4.7.1/zmq_addon.hpp	2020-10-05 05:50:40.000000000 +0800
+++ cppzmq/zmq_addon.hpp	2021-06-18 22:07:35.450858300 +0800
@@ -236,7 +236,7 @@
     message_t encoded(mmsg_size);
     unsigned char *buf = encoded.data<unsigned char>();
     for (const auto &part : parts) {
-        const uint32_t part_size = part.size();
+        const uint32_t part_size = static_cast<uint32_t>(part.size());
         const unsigned char *part_data =
           static_cast<const unsigned char *>(part.data());
 
